### ISSUE TITLE
[FEATURE] Supporting file management in resource center is particularly important（支持资源中心的文件管理尤其重要）

### ISSUE THREAD

**Issue by Narcasserun** [streamxhub/streamx#234]
For the support of resource management, I will give the general function first, and then submit pr
### Resource center function design point

- How to configure resource storage type
- How to upload resources
- How to create a new UDF function

**Comment by Al-assad** [comment]
> For the support of resource management, I will give the general function first, and then submit pr
> 
> ### Resource center function design point
> * How to configure resource storage type
> * How to upload resources
> * How to create a new UDF function

looks like the first 2 point are related to issue#233, we can focus on resource files management of StreamX in issue#233

**Comment by Narcasserun** [comment]
> > 对于资源管理的支持，我先给出通用功能，然后提交pr
> > ### 资源中心功能设计点
> > 
> > * 如何配置资源存储类型
> > * 如何上传资源
> > * 如何创建新的 UDF 函数
> 
> 貌似前2点和issue#233有关，我们可以在issue#233中重点关注StreamX的资源文件管理

issue#233 The discussion is mainly about the streamx itself, and the resource management is faced with the external resources that the user job depends on, such as
- File
- JAR,UDF

**Comment by Narcasserun** [comment]
> > 对于资源管理的支持，我先给出通用功能，然后提交pr
> > ### 资源中心功能设计点
> > 
> > * 如何配置资源存储类型
> > * 如何上传资源
> > * 如何创建新的 UDF 函数
> 
> 貌似前2点和issue#233有关，我们可以在issue#233中重点关注StreamX的资源文件管理

**Comment by Narcasserun** [comment]
issue#233 The discussion is mainly about the streamx itself, and the resource management is faced with the external resources that the user job depends on, such as
- File
- JAR , UDF

**Comment by Al-assad** [comment]
> issue#233 The discussion is mainly about the streamx itself, and the resource management is faced with the external resources that the user job depends on, such as
> 
> * File
> * JAR , UDF

yeah, the focus of our discussion in issue #233 is actually on StreamX's management of the user's external resources as well.
see: [https://github.com/streamxhub/streamx/issues/233#issuecomment-877960948](https://github.com/streamxhub/streamx/issues/233#issuecomment-877960948)
I think the building resource also includes additional user dependencies such as Job JAR, UDF JAR, etc.

**Comment by Narcasserun** [comment]
yeah,I saw it [https://github.com/streamxhub/streamx/issues/233#issuecomment-877960948](url) ,You're right，It's great to let users freely choose the resource storage of streamx，I will realize the upload of external dependent resources, as well as the function of UDF creat, and we can keep communication later

**Comment by Al-assad** [comment]
Just my two cents, plz starting a new issue dedicated to UDF implementation to better focus the discussion.
I'll mark this issue as a duplicate.

**Comment by Al-assad** [comment]
Duplicate of #233

**Comment by Al-assad** [comment]
Duplicate of #233
